src/multipatch/connectivity/edge_transformation.hpp

```diff
@@ -12,6 +12,147 @@
 #include "interface.hpp"
 
 
+/**
+ * @brief Struct to identify the type of an index range on the current 
+ * patch and the index range type on the target patch. 
+ */
+template <class Interface, class PatchOrDim>
+struct IdxRangeCurrentAndTarget;
```

Comment:
This class with its 4 specialisations seems like a lot of code for something that can be achieved with 2 well placed `conditional_t`